# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1349248245 14400
#      Wed Oct 03 03:10:45 2012 -0400
# Node ID c7d7d43f26bba736d06316dcc17d27b0fb9ff663
# Parent  3f447bcc8488525aca724260a45656b75abdf89a
call Ffoo functions instead of setting Vfoo directly.

* octave.cc (octave_process_command_line): Call
Fbuilt_in_docstrings_file instead of setting
Vbuilt_in_docstrings_file directly.  Likewise for Fdoc_cache_file,
Finfo_file, Finfo_program, and Ftexi_macros_file.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -747,22 +747,22 @@ octave_process_command_line (int argc, c
           break;
 
         case 'v':
           print_version_and_exit ();
           break;
 
         case BUILT_IN_DOCSTRINGS_FILE_OPTION:
           if (optarg)
-            Vbuilt_in_docstrings_file = optarg;
+            Fbuilt_in_docstrings_file (octave_value (optarg));
           break;
 
         case DOC_CACHE_FILE_OPTION:
           if (optarg)
-            Vdoc_cache_file = optarg;
+            Fdoc_cache_file (octave_value (optarg));
           break;
 
         case EVAL_OPTION:
           if (optarg)
             {
               if (code_to_eval.empty ())
                 code_to_eval = optarg;
               else
@@ -781,22 +781,22 @@ octave_process_command_line (int argc, c
 
         case IMAGE_PATH_OPTION:
           if (optarg)
             image_path = optarg;
           break;
 
         case INFO_FILE_OPTION:
           if (optarg)
-            Vinfo_file = optarg;
+            Finfo_file (octave_value (optarg));
           break;
 
         case INFO_PROG_OPTION:
           if (optarg)
-            Vinfo_program = optarg;
+            Finfo_program (octave_value (optarg));
           break;
 
         case JIT_DEBUGGING_OPTION:
           Fenable_jit_debugging (octave_value (true));
           break;
 
         case LINE_EDITING_OPTION:
           forced_line_editing = true;
@@ -831,17 +831,17 @@ octave_process_command_line (int argc, c
           break;
 
         case PERSIST_OPTION:
           persist = true;
           break;
 
         case TEXI_MACROS_FILE_OPTION:
           if (optarg)
-            Vtexi_macros_file = optarg;
+            Ftexi_macros_file (octave_value (optarg));
           break;
 
         case TRADITIONAL_OPTION:
           maximum_braindamage ();
           break;
 
         default:
           // getopt_long should print a message about unrecognized
